homeassistant/components/eight_sleep/__init__.py

```diff
@@ -70,17 +63,29 @@ def _get_device_unique_id(eight: EightSleep, user_obj: EightUser | None = None)
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the Eight Sleep component."""
+    """Old set up method for the Eight Sleep component."""
+    if DOMAIN in config:
+        _LOGGER.warning(
+            "Your Eight Sleep configuration has been imported into the UI; "
+            "please remove it from configuration.yaml as support for it "
+            "will be removed in a future release"
+        )
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
 
-    if DOMAIN not in config:
-        return True
+    return True
 
-    conf = config[DOMAIN]
-    user = conf[CONF_USERNAME]
-    password = conf[CONF_PASSWORD]
 
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the Eight Sleep config entry."""
     eight = EightSleep(
-        user, password, hass.config.time_zone, async_get_clientsession(hass)
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        hass.config.time_zone,
+        async_get_clientsession(hass),
     )
 
     hass.data.setdefault(DOMAIN, {})
```

Comment:
Minor nit: Usually we avoid storing data until we know we are going to actually setup (ie passed the coordinator first refresh)